dev/dev_container/Dockerfile

```diff
@@ -0,0 +1,52 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+FROM ubuntu:18.04
+
+# Basic dev dependencies
+RUN apt-get update
+## RUN apt-get -y install build-essential
+# Do we really want such an old version of gcc?
+# apt-cache doesn't list 4.9 for either gcc or g++
+RUN apt-get -y install gcc-4.8
+RUN apt-get -y install g++-4.8
+# compiler ENVs
+ENV CC=gcc-4.8
+ENV CXX=g++-4.8
+# vim (not essential)
+RUN apt-get -y install vim
+# git
+RUN apt-get -y install git
+# wget
+RUN apt-get -y install wget
+# Miniconda (Python 3.6, 64-bit, x86, latest)
+RUN wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+-O mconda.sh
+RUN /bin/bash mconda.sh -b -p miniconda 
+ENV PATH="/miniconda/bin:$PATH"
+
+# create conda env with deps
+RUN conda create -y -q -n pyarrow-dev \
```

Comment:
Run `conda clean --all` afterwards in the same `RUN` step so that the conda cache is not persisted in the image.